packages/pwa-buildpack/lib/WebpackTools/configureWebpack.js

```diff
@@ -90,18 +91,32 @@ async function configureWebpack({ context, vendor = [], special = {}, env }) {
 
     const mode = getMode(env, projectConfig);
 
+    const serviceWorkerFileHash = await createFileHash('./src/sw.js');
+
+    projectConfig.env = {
+        ...projectConfig.env,
+        SERVICE_WORKER_HASH: serviceWorkerFileHash
+    };
+
     const config = {
         mode,
         context, // Node global for the running script's directory
         entry: {
-            client: path.resolve(paths.src, 'index.js')
+            client: path.resolve(paths.src, 'index.js'),
+            registerSW: path.resolve(paths.src, 'registerSW.js')
```

Comment:
What benefits do we get from splitting this out into its own entry point?

Reply:
`registerSW.js` has the registration part for the SW which will never change. So if we separate that out, it will only be requested once in the app's lifetime (assuming cache is not cleared).